### PR TITLE
e2e: create rke2 cluster

### LINKED ISSUE #248: e2e: add test with RKE2
Current version of E2E tests currently create K3s cluster with Elemental, but K3s is in Tech Preview and so not supported for now. Only RKE2 is, so E2E tests should create a RKE2 cluster instead.

### PR DESCRIPTION
**Pull request by ldevulder**
Should fix #248.

Verification run on my lab (limited to 2 nodes, 3 killed my laptop): https://github.com/ldevulder/elemental/actions/runs/2920853424

### CODE REVIEW

**Review by mjura — APPROVED** [review]
For me it looks great, but I would wait for @Itxaka

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -15,13 +15,14 @@ jobs:
     runs-on: kvm-host
     env:
       TIMEOUT_SCALE: 2
-      CLUSTER_NAME: cluster-k3s
+      CLUSTER_NAME: cluster-rke2
       CLUSTER_NS: fleet-default
-      INSTALL_K3S_VERSION: v1.23.9+k3s1
+      K8S_VERSION: v1.23.8+rke2r1
+      INSTALL_K3S_VERSION: v1.23.8+k3s2
```
Comment (juadk) [inline .github/workflows/e2e.yaml:21]: _No need to change something, just a remark here._

I understand why we have `K8S_VERSION` and `INSTALL_K3S_VERSION` but I find it a little bit confusing.
Maybe variable names can be improved, at the end, `INSTALL_K3S_VERSION` is the cluster where Rancher manager is installed and `K8S_VERSION` is the version for the elemental nodes.

File: .github/workflows/e2e.yaml [hunk 2]
```diff
@@ -15,13 +15,14 @@ jobs:
...
       INSTALL_K3S_SKIP_ENABLE: true
       K3S_KUBECONFIG_MODE: 0644
       KUBECONFIG: /etc/rancher/k3s/k3s.yaml
-      PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+      #PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```
Comment (juadk) [inline .github/workflows/e2e.yaml:25]: Can this comment be removed (or explained)? leftover?

File: .github/workflows/e2e.yaml (hunk 1 above)
Comment (ldevulder) [reply .github/workflows/e2e.yaml:21]: `INSTALL_K3S_VERSION` cannot be changed as this variable is used by the K3s installer. But I will rename `K8S_VERSION` and also add comments to explain these variables.

File: .github/workflows/e2e.yaml (hunk 2 above)
Comment (ldevulder) [reply .github/workflows/e2e.yaml:25]: Good catch! Not needed anymore yes, simply forgot to remove it.

**Comment by ldevulder** [comment]
> For me it looks great, but I would wait for @Itxaka

He is an vacation until this end of the month, I have other changes to do in the E2E tests, so I will merge this PR (verification run is green). And I will change after if he add some comments/changes.

**Review by juadk — APPROVED** [review]

**Pull request merged**